src/InMemoryApp.js

```diff
@@ -0,0 +1,9 @@
+import App from "./App";
+import {useState} from "react";
+
+function InMemoryApp(props) {
+    const [data, setData] = useState(props.initialData);
+    return <App data={data} setData={setData}/>
```

Comment:
I suggest that you don't provide setData to the App. Instead, provide things like onItemChanged, onItemAdded, onItemDeleted.  Move the code for those from App to here.  That way, App isn't involved with maintaining the state; it's all localized here.